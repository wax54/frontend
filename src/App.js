import NavBar from './NavBar';
import Routes from './Routes';
import UserManager from './User/UserManager';

import './App.css';

function App() {
  return (
    <div className="App">
      <UserManager>
        <NavBar />
        <Routes />
      </UserManager>
    </div>
  );
}

export default App;
