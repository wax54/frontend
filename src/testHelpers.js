import { render, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import UserContext from './UserContext';

const validUserContext = {
    user: {
        username: "wax54",
        firstName: "Sam",
        lastName: "Crewe-Sullam",
        email: "wax654@gmail.c om",
        isAdmin: false,
        applications: [1, 2, 3]
    },
    isLoggedIn: true
}

function preRender(options={}) {
    const defaultOptions = {
        initialEntries: ['/'],
        context: {
            user: {},
            isLoggedIn: false
        }
    }
    const newOptions = {
        initialEntries: 
            [...(options.initialEntries || []), ...defaultOptions.initialEntries],
        context: 
            { ...defaultOptions.context, ...(options.context || {}) }
    }
    //console.log('options: ', options, 'newOptions', newOptions);
    return newOptions;
}

function renderWithRouter(Component, options) {
    options = preRender(options);
    return render(
        <MemoryRouter initialEntries={options.initialEntries}>
            <Component />
        </MemoryRouter>
    );
}

async function renderAndWaitFor(Component, options) {
    options = preRender(options);
    return await waitFor(() => render(
        <MemoryRouter initialEntries={options.initialEntries}>
            <Component />
        </MemoryRouter>
    ));
}


function renderWithUserContext(Component, options) {
    options = preRender(options);
    return render(
        <MemoryRouter initialEntries={options.initialEntries}>
            <UserContext.Provider value={options.context} >
                <Component />
            </UserContext.Provider>
        </MemoryRouter>
    );
}

async function renderWithUserContextAndWait(Component, options) {
    options = preRender(options);
    return await waitFor(() => render(
        <MemoryRouter initialEntries={options.initialEntries}>
            <UserContext.Provider value={options.context} >
                <Component />
            </UserContext.Provider>
        </MemoryRouter>
    ));
}



export { validUserContext, 
    renderWithRouter, 
    renderAndWaitFor, 
    renderWithUserContext, 
    renderWithUserContextAndWait }