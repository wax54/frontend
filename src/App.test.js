import { renderAndWaitFor} from './testHelpers';
import App from './App';

test("doesn't blow up", async () => {
    await renderAndWaitFor(App);
});

test("matchesSnapshot", async () => {
    const { asFragment } = await renderAndWaitFor(App);
    expect(asFragment()).toMatchSnapshot();
});
