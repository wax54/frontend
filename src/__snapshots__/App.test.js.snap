// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchesSnapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="NavBar"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Home
      </a>
      <a
        href="/login"
      >
         login
      </a>
      <a
        href="/signup"
      >
        Signup
      </a>
    </div>
    <div
      class="Home"
    >
      <h1>
        Please Login or Signup above!
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
